src/EnergyPlus/DataSizing.cc

```diff
@@ -447,86 +445,68 @@ void GetCoilDesFlowT(EnergyPlusData &state,
     // energy and mass flow balance
 
     // FUNCTION LOCAL VARIABLE DECLARATIONS:
-    int DDAtSensPeak(0);
-    int TimeStepAtSensPeak(0);
-    int DDAtFlowPeak(0);
-    int TimeStepAtFlowPeak(0);
-    int CoolCapCtrl; // type of coil capacity control
-    int PeakLoadType;
-    int DDAtTotPeak(0);
-    int TimeStepAtTotPeak(0);
-    int TimeStepAtPeak(0);
-    Real64 ZoneCoolLoadSum(0); // sum of zone cooling loads at the peak [W]
-    Real64 AvgZoneTemp(0);     // average zone temperature [C]
-    Real64 AvgSupTemp(0.0);    // average supply temperature for bypass control [C]
-    Real64 TotFlow(0.0);       // total flow for bypass control [m3/s]
-    Real64 MixTemp(0.0);       // mixed air temperature at the peak [C]
-
-    auto &SysSizInput(state.dataSize->SysSizInput);
-    auto &FinalSysSizing(state.dataSize->FinalSysSizing);
-    auto &SysSizPeakDDNum(state.dataSize->SysSizPeakDDNum);
-    auto &CalcSysSizing(state.dataSize->CalcSysSizing);
-
-    CoolCapCtrl = SysSizInput(SysNum).CoolCapControl;
-    PeakLoadType = SysSizInput(SysNum).CoolingPeakLoadType;
-    DDAtSensPeak = SysSizPeakDDNum(SysNum).SensCoolPeakDD;
-    if (DDAtSensPeak > 0) {
-        TimeStepAtSensPeak = SysSizPeakDDNum(SysNum).TimeStepAtSensCoolPk(DDAtSensPeak);
-        DDAtFlowPeak = SysSizPeakDDNum(SysNum).CoolFlowPeakDD;
-        TimeStepAtFlowPeak = SysSizPeakDDNum(SysNum).TimeStepAtCoolFlowPk(DDAtFlowPeak);
-        DDAtTotPeak = SysSizPeakDDNum(SysNum).TotCoolPeakDD;
-        TimeStepAtTotPeak = SysSizPeakDDNum(SysNum).TimeStepAtTotCoolPk(DDAtTotPeak);
-
-        if (PeakLoadType == TotalCoolingLoad) {
-            TimeStepAtPeak = TimeStepAtTotPeak;
+    int TimeStepAtPeak = 0;
+    Real64 ZoneCoolLoadSum = 0; // sum of zone cooling loads at the peak [W]
+    Real64 AvgZoneTemp = 0;     // average zone temperature [C]
+
+    auto &finalSysSizing = state.dataSize->FinalSysSizing(SysNum);
+    auto &sysSizPeakDDNum = state.dataSize->SysSizPeakDDNum(SysNum);
+    auto &calcSysSizing = state.dataSize->CalcSysSizing(SysNum);
+
+    int sysSizIndex =
+        UtilityRoutines::FindItemInList(finalSysSizing.AirPriLoopName, state.dataSize->SysSizInput, &SystemSizingInputData::AirPriLoopName);
+    if (sysSizIndex == 0) sysSizIndex = 1;
+    auto &sysSizInput = state.dataSize->SysSizInput(sysSizIndex);
```

Comment:
Basic correction to issue. SysSizInput and FinalSysSizing arrays will have different index when the Sizing:System and AirloopHVAC objects are not in the same order in idf.

Reply:
Good catch.